projects/HVAC/vent_control/vent.ino

```diff
@@ -0,0 +1,260 @@
+/*
+
+motor: 28BYJ48-5V
+  steps: 8
+  stride angle: 5.625/64
+
+  loops to 1 revolution: 360/(8 * 5.625/64) = 512
+
+            | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |
+  4 orange  | x | x |   |   |   |   |   | x |
+  ----------|---|---|---|---|---|---|---|---|
+  3 yellow  |   | x | x | x |   |   |   |   |
+  ----------|---|---|---|---|---|---|---|---|
+  2 pink    |   |   |   | x | x | x |   |   |
+  ----------|---|---|---|---|---|---|---|---|
+  1 blue    |   |   |   |   |   | x | x | x |
+  ----------|---|---|---|---|---|---|---|---|
+
+Example:
+
+@VENT01>set degrees: 90
+@VENT01>set percents: 25
+
+*/
+
+#include <SyrotaAutomation1.h>
+
+void motor_step_cw();
+void motor_step_ccw();
+bool motor_zero();
+void motor_rotate_degrees(int degrees);
+void motor_rotate_percents(int percents);
+void motor_rotate_to(int position);
+
+void set_error(char *message);
+
+// Handles:
+//   * set degrees: -360..360
+//   * set percents: -100..100
+void process_command();
+
+int motor_position = 0;
+// steps per rotation
+const int motor_steps_per_rotation = 512;
+const float motor_steps_per_degree = motor_steps_per_rotation / 360.0;
+
+const int motor_sensor_pin = 2;
+const int blink_pin = 13;
+
+const int serial_pin = 3;
+SyrotaAutomation net = SyrotaAutomation(serial_pin);
+const char* net_address = "VENT01";
+
+
+void setup() {
```

Comment:
Check out config implementation here: https://github.com/sergesyrota/Sprinkler-controller/tree/master/Arduino
in config.h: struct configuration_t describes the structure. Arduino.ino#18 has defaults. And then in Arduino.ino#32, I'm reading config from persistent memory. If nothing's there, or version mismatch, I discard and write defaults. This way we can maintain persistent configuration for things that can change over time. Although note that EEPROM has very limited read/write cycles :)